test/edu/washington/escience/myria/operator/SplitTest.java

```diff
@@ -0,0 +1,167 @@
+package edu.washington.escience.myria.operator;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.regex.PatternSyntaxException;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.primitives.Longs;
+
+import edu.washington.escience.myria.DbException;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.storage.TupleBatch;
+import edu.washington.escience.myria.storage.TupleBatchBuffer;
+import edu.washington.escience.myria.util.TestEnvVars;
+
+public class SplitTest {
+
+  @Test
+  public void testGeneratedSplits() throws DbException {
+    final Object[][] expectedResults =
+        { {true, "foo", 1L, 0.1}, {true, "bar", 1L, 0.1}, {true, "baz", 1L, 0.1,},
+            {false, "", 2L, 0.2}, {false, "qux", 2L, 0.2}, {false, "", 2L, 0.2},
+            {false, "", 2L, 0.2}};
+    final Schema schema =
+        new Schema(ImmutableList.of(Type.BOOLEAN_TYPE, Type.STRING_TYPE, Type.LONG_TYPE,
+            Type.DOUBLE_TYPE), ImmutableList.of("bool", "string", "long", "double"));
+    final TupleBatchBuffer input = new TupleBatchBuffer(schema);
+    // First row to explode
+    input.putBoolean(0, true);
+    input.putString(1, "foo:bar:baz");
+    input.putLong(2, 1L);
+    input.putDouble(3, 0.1);
+    // Second row to explode
+    input.putBoolean(0, false);
+    input.putString(1, ":qux::");
+    input.putLong(2, 2L);
+    input.putDouble(3, 0.2);
+    Split splitOp = new Split(new TupleSource(input), 1, ":");
+
+    splitOp.open(TestEnvVars.get());
+    TupleBatch result;
+    int rowIdx = 0;
+    while (!splitOp.eos()) {
+      result = splitOp.nextReady();
+      if (result != null) {
+        assertEquals(schema.numColumns(), result.getSchema().numColumns());
+        assertEquals(Type.BOOLEAN_TYPE, result.getSchema().getColumnType(0));
```

Comment:
why not just assert that `schema.getColumnTypes().equals(resultSchema.getColumnTypes())`?